packages/amplify-category-api/src/__tests__/provider-utils/awscloudformation/cfn-api-artifact-handler.test.ts

```diff
@@ -357,4 +357,36 @@ describe('update artifacts', () => {
       },
     ]);
   });
+
+  it('correctly updates the cli-inputs on an update that sets a NEW lambda for conflict resolution', async () =>{
+    jest.spyOn(AppsyncApiInputState.prototype, 'saveCLIInputPayload');
+    jest.spyOn(AppsyncApiInputState.prototype, 'cliInputFileExists').mockReturnValueOnce(true);
+
+    updateRequestStub.serviceModification.conflictResolution = {
+      defaultResolutionStrategy: {
+        type: 'LAMBDA',
+        resolver: {
+          type: 'NEW',
+        },
+      },
+    };
+
+    await cfnApiArtifactHandler.updateArtifacts(updateRequestStub);
+
+    const { objectContaining, stringContaining } = expect;
+
+    expect(AppsyncApiInputState.prototype.saveCLIInputPayload).toHaveBeenCalledWith(objectContaining({
+      serviceConfiguration: objectContaining({
+        conflictResolution: {
+          defaultResolutionStrategy: {
+            type: 'LAMBDA',
+            resolver: {
+              type: 'EXISTING',
```

Comment:
I see that we're passing in the type as `NEW` in the update request and it's getting stored here as `EXISTING`, is this intentional?

Reply:
Yeah, the behavior is slightly confusing, but in the call to `updateArtifacts` we actually check if the value is NEW, generate a new lambda, and then update the value to EXISTING w/ the generated values.